iceoryx_utils/test/moduletests/test_cxx_expected.cpp

```diff
@@ -82,6 +82,15 @@ TEST_F(expected_test, CreateError)
     EXPECT_THAT(sut.get_error().m_b, Eq(212));
 }
 
+TEST_F(expected_test, CreateValueAndMove)
```

Comment:
Please name the test like: What are you doing at what are you expecting from the public api or the behavior.
In this case `MoveAssignmentInvalidatesSource`

Additionally, you are missing some tests:
1. expected with value and error
    1. move ctor of expected with value invalidates source
    1. move ctor of expected with error invalidates source
    1. move assignment of expected with value invalidates source
    1. move assignment of expected with error invalidates source
1. expected with only error
    1. move ctor of expected with error invalidates source
    1. move assignment of expected with error invalidates source

Reply:
Well, well, I remember a certain young @elfenpiff writing these tests some years ago :stuck_out_tongue: Reworking existing ones and will work on the ones you mentioned.

Reply:
Yeah yeah ... I was young, naive and didn't knew what I was doing. But then Hypnotoad came and showed me the road to enlightenment.
![thatsnoneofmy](https://user-images.githubusercontent.com/56729169/107778723-b792ff80-6d44-11eb-8e26-9e612c32d32f.jpg)